test/EntityFramework/UnitTests/Migrations/Infrastructure/EdmModelDifferTests.cs

```diff
@@ -3650,6 +3654,93 @@ var operations
             Assert.Equal(0, operations.Count);
         }
 
+        public interface IPlatformEntity
+        {
+            long Id { get; set; }
+        }
+
+        [Table("PlatformUser")]
+        public class PlatformUser : IPlatformEntity
+        {
+            public long Id { get; set; }
+
+            public string UserName { get; set; }
+        }
+
+        public class TypeGenerator
```

Comment:
If the extension to the model is compiled the model generator finds it and builds the initial model with the extension already in it. I needed the initial model to be created without the extension, then add the extension as part of a migration, then perform the comparison between two models. This matches the scenario of deliver a database, then update that database with a new model later. The scenario we are getting the exception while doing now.